rules/no-jasmine-globals.js

```diff
@@ -17,29 +26,20 @@ module.exports = {
         if (!calleeName) {
           return;
         }
+        if (
+          calleeName === 'spyOn' ||
+          calleeName === 'spyOnProperty' ||
+          calleeName === 'fail' ||
+          calleeName === 'pending'
+        ) {
+          if (scopeHasLocalReference(context.getScope(), calleeName)) {
+            // It's a local variable, not a jasmine global.
+            return;
+          }
 
-        if (calleeName === 'spyOn' || calleeName === 'spyOnProperty') {
-          context.report({
-            node,
-            message: `Illegal usage of global \`${calleeName}\`, prefer \`jest.spyOn\``,
-          });
-          return;
-        }
-
-        if (calleeName === 'fail') {
-          context.report({
-            node,
-            message:
-              'Illegal usage of `fail`, prefer throwing an error, or the `done.fail` callback',
-          });
-          return;
-        }
-
-        if (calleeName === 'pending') {
           context.report({
             node,
-            message:
-              'Illegal usage of `pending`, prefer explicitly skipping a test using `test.skip`',
+            message: globalAlternatives[calleeName],
```

Comment:
can we use `messageId` instead?

https://eslint.org/docs/developer-guide/working-with-rules#messageids

Reply:
and `data` for interpolation

Reply:
Sure.  Something as simple as this, or would you rather try & refactor to use message-interpolation and fewer separate messageIds?

```diff
diff --git a/rules/no-jasmine-globals.js b/rules/no-jasmine-globals.js
index e081a4b..d7794fe 100644
--- a/rules/no-jasmine-globals.js
+++ b/rules/no-jasmine-globals.js
@@ -2,21 +2,21 @@
 
 const { getDocsUrl, getNodeName, scopeHasLocalReference } = require('./util');
 
-const globalAlternatives = {
-  spyOn: 'Illegal usage of global `spyOn`, prefer `jest.spyOn`',
-  spyOnProperty: 'Illegal usage of global `spyOnProperty`, prefer `jest.spyOn`',
-  fail:
-    'Illegal usage of `fail`, prefer throwing an error, or the `done.fail` callback',
-  pending:
-    'Illegal usage of `pending`, prefer explicitly skipping a test using `test.skip`',
-};
-
 module.exports = {
   meta: {
     docs: {
       url: getDocsUrl(__filename),
     },
     fixable: 'code',
+    messages: {
+      spyOn: 'Illegal usage of global `spyOn`, prefer `jest.spyOn`',
+      spyOnProperty:
+        'Illegal usage of global `spyOnProperty`, prefer `jest.spyOn`',
+      fail:
+        'Illegal usage of `fail`, prefer throwing an error, or the `done.fail` callback',
+      pending:
+        'Illegal usage of `pending`, prefer explicitly skipping a test using `test.skip`',
+    },
   },
   create(context) {
     return {
@@ -39,7 +39,7 @@ module.exports = {
 
           context.report({
             node,
-            message: globalAlternatives[calleeName],
+            messageId: calleeName,
           });
           return;
         }
```

Reply:
>or would you rather try & refactor to use message-interpolation and fewer separate messageIds?

yes please 🙂

Reply:
at least for `spyOn`, which are very similar

Reply:
See what you think to https://github.com/jest-community/eslint-plugin-jest/pull/205/commits/6d7c3056c417ac8ff768ac35df919d0bb77d8e1b